File: Java/Sorting /Selection_sort.java

/**
 * @author Kritika Srivastava
 * @since October 09, 2021
 *
 * Program Description - Selection sort Code
 */
import java.util.Scanner;

public class Selection_sort {
    public static void main(String[] args) {
        Scanner kb = new Scanner(System.in);

        // Input
        System.out.println("Enter size of array:");
        int n = kb.nextInt(); // Size of array
        int arr[] = new int[n];
        System.out.println("Enter array elements:");
        for (int i = 0; i < n; i++) {
            arr[i] = kb.nextInt();
        }

        // Selection sort Algorithm

        for (int i = 0; i < n; i++) {
            int min = i;
            for (int j = i + 1; j < n; j++) {
                if (arr[j] < arr[min])
                    min = j;
            }
            int t = arr[i];
            arr[i] = arr[min];
            arr[min] = t;
        }
        System.out.println("The numbers arranged in ascending order(selection sort) are:");
        for (int i = 0; i < n; i++) {
            System.out.println(arr[i]);
        }

    }
}

File: SelectionSort.cpp
#include <iostream>
using namespace std;
int main()
{
    int n;
    cout<<"Enter the size of Array\n";
    cin>>n;
    int a[n];
    cout<<"Enter the elements\n";
    for(int i=0;i<n;i++){
        cin>>a[i];
    }
    
    for(int i=0;i<n-1;i++){
        for(int j=i+1;j<n;j++){
            if(a[j]<a[i]){
                int t=a[j];
                a[j]=a[i];
                a[i]=t;
            }
        }
    }
    cout<<"The sorted Array\n";
    for(int i=0;i<n;i++){
        cout<<a[i]<<endl;
    }
    return 0;
}
